PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['hr.applicant']
-    _inherit = ['hr.candidate']
-    _inherit = ["hr.job"]

Total Changes: 7

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment_skills/models/hr_applicant.py
+++ b/addons/hr_recruitment_skills/models/hr_applicant.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class HrApplicant(models.Model):
-    _inherit = ['hr.applicant']
+    _inherit = 'hr.applicant'
 
     candidate_skill_ids = fields.One2many(related="candidate_id.candidate_skill_ids", readonly=False)
     skill_ids = fields.Many2many(related="candidate_id.skill_ids", readonly=False)

--- a/addons/hr_recruitment_skills/models/hr_candidate.py
+++ b/addons/hr_recruitment_skills/models/hr_candidate.py
@@ -6,7 +6,7 @@ from odoo import fields, models, api
 
 
 class HrCandidate(models.Model):
-    _inherit = ['hr.candidate']
+    _inherit = 'hr.candidate'
 
     candidate_skill_ids = fields.One2many('hr.candidate.skill', 'candidate_id', string="Skills")
     skill_ids = fields.Many2many('hr.skill', compute='_compute_skill_ids', store=True)

--- a/addons/hr_recruitment_skills/models/hr_candidate_skill.py
+++ b/addons/hr_recruitment_skills/models/hr_candidate_skill.py
@@ -5,6 +5,7 @@ from odoo.exceptions import ValidationError
 
 
 class HrCandidateSkill(models.Model):
+    _name = 'hr.candidate.skill'
     _description = "Skill level for a candidate"
     _rec_name = 'skill_id'
     _order = "skill_level_id"

--- a/addons/hr_recruitment_skills/models/hr_job.py
+++ b/addons/hr_recruitment_skills/models/hr_job.py
@@ -5,7 +5,7 @@ from odoo import fields, models, _
 
 
 class HrJob(models.Model):
-    _inherit = ["hr.job"]
+    _inherit = "hr.job"
 
     skill_ids = fields.Many2many(comodel_name='hr.skill', string="Expected Skills")
 
